Fix access to ieee802154 APIs on nRF5340 application core

jciupis opened the pull request.

Commit:
dts: arm: nordic: add ieee802154 node to nrf5340_cpuapp

This commit adds an ieee802154 node to the list of nRF5340 application
core's peripherals. While it does not translate directly into a physical
RADIO peripheral, it represents the capability to use the ieee802154
radio (indirectly, through the network core).

Signed-off-by: Jędrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

Commit:
boards: arm: nordic: add ieee802154 to chosen for nrf5340_cpuapp

This commit adds ieee802154 node to chosen for boards based on nRF5340
application core.

Signed-off-by: Jędrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

gmarull commented on dts/arm/nordic/nrf5340_cpuapp_peripherals.dtsi:
```diff
@@ -507,3 +507,7 @@ gpio1: gpio@842800 {
 	port = <1>;
 };
+
+ieee802154: ieee802154 {
```
needs to be a child of `nordic,nrf-radio`

gmarull replied:
nevermind, I see that radio is driven by network core.

Commit:
boards: arm: nordic: add ieee802154 to chosen for nrf5340_cpunet

This commit adds ieee802154 node to chosen for boards based on nRF5340
network core.

Signed-off-by: Jędrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

jciupis commented on boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_common.dts:
```diff
@@ -14,6 +14,7 @@
 		zephyr,bt-mon-uart = &uart0;
 		zephyr,bt-c2h-uart = &uart0;
 		zephyr,bt-hci-rpmsg-ipc = &ipc0;
+		zephyr,ieee802154 = &ieee802154;
```
It's fixed for the network core as well now.

Commit:
samples: net: wpan(usb|serial): exclude thingy53_nrf5340_cpuapp_ns

This commit excludes thingy53_nrf5340_cpuapp_ns platform from building
wpanusb and wpan_serial samples. The platform doesn't support TF-M yet
and therefore the build is bound to fail.

Due to limitations of the filter functionality, a more generic approach
where _ns platforms without TF-M support would be excluded from the
build, for example:

filter: dt_chosen_enabled("zephyr,ieee802154") and
	CONFIG_NRF_SOC_SECURE_SUPPORTED

doesn't properly filter out the needed platforms. As a result,
platform_exclude syntax is used to specifically exclude the platform.

Signed-off-by: Jędrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

anangl commented on samples/net/wpan_serial/sample.yaml:
```diff
@@ -8,7 +8,7 @@ common:
 tests:
   sample.net.wpan.serial:
-    filter: dt_chosen_enabled("zephyr,ieee802154")
+    filter: dt_chosen_enabled("zephyr,ieee802154") and (not CONFIG_TRUSTED_EXECUTION_NONSECURE or CONFIG_TFM_BOARD!="")
```
Perhaps this would be clearer:
```suggestion
    filter: dt_chosen_enabled("zephyr,ieee802154") and (not CONFIG_TRUSTED_EXECUTION_NONSECURE or CONFIG_BUILD_WITH_TFM)
```

jciupis replied:
I replaced the expression with a single, already existing symbol
